docker-compose.yml

```diff
@@ -22,7 +25,12 @@ services:
       target: development
     environment:
       # Be sure to add environment variables to config/options.yml
-      - DATABASE_URL=postgres://app:secret@postgres/app_development
+      - HOSTEDGPT_DB_USERNAME=${HOSTEDGPT_DB_USERNAME:-hostedgpt_user}
+      - HOSTEDGPT_DB_PASSWORD=${HOSTEDGPT_DB_PASSWORD:-secret}
+      - HOSTEDGPT_DB_HOST=${HOSTEDGPT_DB_HOST:-postgres}
+      - HOSTEDGPT_DB_PORT=${HOSTEDGPT_DB_PORT:-5432}
+      - HOSTEDGPT_DB_NAME=${HOSTEDGPT_DB_NAME:-hostedgpt}
+      - DATABASE_URL
```

Comment:
if DATABASE_URL is set then database.yml will use that as the url option (which will override other settings for the db there), otherwise we use the env vars here (which have default values now)

Reply:
Ah, I definitely see that we left off the HOSTEDGPT_DB_* env names from this compose file and needed to add those. That’s a good catch.

But why not continue to provide the defaults through the DATABASE_URL one? I think it effectively works the same, but I leaned slightly towards using DATABASE_URL because if we do that we’re subtly teaching people that this is what they should change locally if they want to configure things differently. Whereas by moving the defaults to all the individual env variables, we’re suggesting to people that they should set these variables locally if they want to change things.

Also, if I was running docker locally successfully yesterday, and then I pull down main and restart docker, will it blow up because the database name and username just changed in this commit? I like the default values yo used better than what was there, but I’m just a little worried that changing this will break it for active users.

Reply:
Oh wait, I just read in another comment that if we use DATABASE_URL and want to change the username and password, then we end up needing to change it in two places? I thought DATABASE_URL became the default so we just had to change it in there, but I may not fully understand how these configurations are interacting.

Reply:
Okay, maybe what you’re doing is that if someone was using HOSTEDGPT_DEV_DB then if we switch our config to use DATABASE_URL it would override their configuration. So you changed docker config to use the individual pieces so as to respect people who have configured their local app using those.

Reply:
yeah the database default name change will break the existing deployments until the new env vars are set up for them. We can go back to the old values if that's a concern. I was thinking it would be less confusing for someone with a local postgres setup to see the "hostedgpt_..." named values